### PR TITLE
Xqueryextensions

### PR DESCRIPTION
**Pull request by BoukeNijhuis**
Dear sir/madam,

The goal was to have the same options for an XQuery Match as we have for a an XPath Match (ignore namespace prefix and ignore comments). The easiest way to reach this goal was to reuse the code in XPathContainsAssertion. Therefore I moved all common code (between XPathContainsAssertion and XQueryContainsAssertion) to a new class (XBaseContainsAssertion).

Unfortunatily it seems the entire AssertionConfigurationDialog.java file is changed, but I have only changed two things:
- it works with an XBaseContainsAssertion instead of an XPathContainsAssertion,
- little UI change to have a more evenly split SplitContainer.

Furthermore I fixed an UI bug regarding ignoreComments. The UI did not properly reflect this setting.

I did not add any testcase to the XPathContainsTest, because it functionality did not change. I only moved functionality between classes (and did some optimizations). For the XQueryMatch I have added new test cases that match the new functionality.

Probably some more code can be moved to the XBaseContainsAssertion, but I think this is proper first step.

I hope you like the change.

Best regards,
Bouke Nijhuis

### CODE REVIEW

**Comment by spindelmanne** [comment]
Seems like a nice contribution. Thanks!

I suspect that the whole-file diff for AssertionConfigurationDialog.java is because you have a line ending difference. Could you see of you can get rid of this issue, so that we can see exactly what you've changed?

**Comment by BoukeNijhuis** [comment]
Fixed the EOL issue.

File: soapui/src/main/java/com/eviware/soapui/impl/wsdl/teststeps/assertions/basic/XBaseContainsAssertion.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+import com.eviware.soapui.model.testsuite.TestProperty;
+import com.eviware.soapui.support.StringUtils;
+import com.eviware.soapui.support.Tools;
+import com.eviware.soapui.support.components.JXToolBar;
+import com.eviware.soapui.support.types.StringList;
+import com.eviware.soapui.support.xml.XmlObjectConfigurationBuilder;
+import com.eviware.soapui.support.xml.XmlObjectConfigurationReader;
+import com.eviware.soapui.support.xml.XmlUtils;
+
+public abstract class XBaseContainsAssertion extends WsdlMessageAssertion implements RequestAssertion, ResponseAssertion,
```
Comment (spindelmanne) [inline soapui/src/main/java/com/eviware/soapui/impl/wsdl/teststeps/assertions/basic/XBaseContainsAssertion.java]: I would choose a different name for this assertion, such as AbstractXmlContainsAssertion. This is more consistent with the naming in the SoapUI codebase, and XBase doesn't really mean anything (nor "X").

**Comment by BoukeNijhuis** [comment]
Dear Manne,

What is the next step?

Best regards,
Bouke

**Comment by spindelmanne** [comment]
Sorry for the late reply.

I'm discussing with the Product owner for SoapUI. Meanwhile, could you try to resolve the conflicts? It can't be merged in it's current state, apparently.

**Comment by spindelmanne** [comment]
When it's mergeable again I'll demonstrate it to the PO and get him to confirm the change. I think it looks good, but it's not my call.

**Comment by BoukeNijhuis** [comment]
It is mergeable again.

**Comment by spindelmanne** [comment]
Hi, we've decided to merge this directly after the next release, which is due next week. We don't want to do it earlier because it'd disrupt the test plan.

**Comment by BoukeNijhuis** [comment]
Dear Manne,

Any updates?

Also regarding my other pull requests?

Best regards,
Bouke

**Pull request merged**